Splitio.Redis/Services/Client/Classes/RedisClient.cs

```diff
@@ -51,10 +53,21 @@ public override void Destroy()
         {
             if (_statusManager.IsDestroyed()) return;
 
-            _uniqueKeysTracker.Stop();
-            _impressionsCounter.Stop();
-            _connectionPoolManager.Dispose();
             base.Destroy();
+
+            _log.Info("Initialitation sdk destroy.");
+
+            var task = new List<Task>
+            {
+                _uniqueKeysTracker.StopAsync(),
+                _impressionsCounter.StopAsync()
+            };
+
+            Task.WaitAll(task.ToArray(), Constants.Gral.DestroyTimeount);
+
+            _connectionPoolManager.Dispose();
+
+            _log.Info("SDK has been destroyed.");
```

Comment:
Maybe something more related to the class where it is? Stopped Redis submitter tasks?